community/cypher/cypher-compiler-2.3/src/test/scala/org/neo4j/cypher/internal/compiler/v2_3/birk/CodeGeneratorTest.scala

```diff
@@ -422,23 +420,69 @@ class CodeGeneratorTest extends CypherFunSuite with LogicalPlanningTestSupport {
     val plan = ProduceResult(List.empty, List.empty, List("a"), Projection(SingleRow()(solved), Map("a" -> SignedDecimalIntegerLiteral("1")(null)))(solved))
 
     // when
-    val completion = mock[CompletionListener]
-    val compiled = compile(plan, completion = completion)
+    val closer = mock[TaskCloser]
+    val compiled = compile( plan, taskCloser = closer )
 
     // then
-    verifyZeroInteractions(completion)
+    verifyZeroInteractions(closer)
     val visitor = mock[ResultVisitor[RuntimeException]]
     val exception = new scala.RuntimeException()
     when(visitor.visit(any[ResultRow])).thenThrow(exception)
     intercept[RuntimeException] {
       compiled.accept(visitor)
-    } // should be(exception) -- TODO: this verification fails, but that failure is unrelated to the current fix...
-    verify(completion).complete(success = false)
+
+    }
+    verify(closer).close(success = false)
   }
 
-  private def compile(plan: LogicalPlan, params: Map[String, AnyRef] = Map.empty, completion: CompletionListener = NOOP) = {
+  test("should throw the same error as the user provides") {
+    // given
+    val plan = ProduceResult(List.empty, List.empty, List("a"), Projection(SingleRow()(solved), Map("a" -> SignedDecimalIntegerLiteral("1")(null)))(solved))
+
+    // when
+    val closer = mock[TaskCloser]
+    val compiled = compile( plan, taskCloser = closer )
+
+    // then
+    val visitor = mock[ResultVisitor[RuntimeException]]
+    val exception = new scala.RuntimeException()
+    when(visitor.visit(any[ResultRow])).thenThrow(exception)
```

Comment:
we are testing that if a user sends in a `ResultsVisitor<MyAwesomeException>` and that visitor throws we should propagate that exception to the user and not hide it. The outer (java) signature is something like

```
<E extends Exception> void accept( ResultVisitor<E> visitor )
            throws E;
```
thus we must propagate `E`